backend/ibutsu_server/tasks/runs.py

```diff
@@ -23,38 +23,35 @@ def _copy_column(result, run, key):
 def update_run(run_id):
     """Update the run summary from the results, this task will retry 1000 times"""
     with lock(f"update-run-lock-{run_id}"):
-        key_map = {
-            "failed": "failures",
-            "error": "errors",
-            "skipped": "skips",
-            "xpassed": "xpasses",
-            "xfailed": "xfailures",
-        }
         run = Run.query.get(run_id)
         if not run:
             return
-        # sort according to starttime to get the most recent starting time of the run
+        # sort according to start_time to get the most recent starting time of the run
         results = (
             Result.query.filter(Result.run_id == run_id).order_by(Result.start_time.asc()).all()
         )
-        summary = {"errors": 0, "failures": 0, "skips": 0, "tests": 0, "xpasses": 0, "xfailures": 0}
-        run_duration = 0.0
+        summary = {
+            "errors": run.summary.get("errors", 0),
+            "failures": run.summary.get("failures", 0),
+            "skips": run.summary.get("skips", 0),
+            "tests": run.summary.get("tests", 0),
+            "xpasses": run.summary.get("xpasses", 0),
+            "xfailures": run.summary.get("xfailures", 0),
+            "tests_collected": run.summary.get("tests_collected", 0),
+        }
```

Comment:
Since you're changing this, can you add in "passes" too? I'm tired of calculating the passes on the fly every time.

Reply:
@rsnyman we would need to calculate the `passes` in the pytest_plugin in order to do that. I can open a PR for that if you'd like.

Reply:
I think we should do it in the plugin too. Doing it here will make it backward compatible, which is not a bad thing :-)